### PR TITLE
Rewrite to have individual aliases for each command

### LINKED ISSUE #21: Clean up use of interfaces in NpmRunner
The use of the `INpmRunnerCommands` and `INpmRunnerConfiguration` interfaces seems to be quite inconsistent. 

While `INpmRunnerConfiguration.FromPath` returns an `INpmRunnerCommands`, `INpmRunnerConfiguration. WithLogLevel` returns an `INpmRunnerConfiguration`. IMHO it should be possible to write either `Npm.FromPath("C:\foo").WithLogLevel(NpmLogLevel.Silent).Install()` or `Npm.WithLogLevel(NpmLogLevel.Silent).FromPath("C:\foo").Install()`.

Also it is not clear to me why the methods of `INpmRunnerCommands` return an `INpmRunnerCommands` object and not `void` or the output of the specific npm command?

One last thing is that there is an `NpmRunner.Pack`, but no `INpmRunnerCommands.Pack`, which makes it impossible to use with the `INpmRunnerConfiguration` functions.

If there's interested in having this cleaned up I can write a PR.

### PR DESCRIPTION
**Pull request by pascalberger**
Here's my approach for cleaning up interfaces and aliases.

I removed the old implementation completely. Maybe it would be possible to keep it and call the new interface from it, but most probably it will lead to a lot of namespace and type clashes anyway.

All in all it should have the same possibilities as the latest (0.8.0) release, but from the API it is a complete breaking change. IMHO this shouldn't be that big a problem since used add-ins can (should) be versioned in the build script.

Some stuff which I've indentionally changed:

* `Force` flag is now part of `NpmInstallSettings` and no longer of the base `NpmSettings`
* `LogLevel` is now part of the base `NpmSettings`
* Different commands are split into their own namespace, as it is done with eg. NuGet or Chocolatey in Cake.Core
* Test cases are built similar to the one in Cake.Core
* I added `NpmLogLevel.Default` which means that it uses log level from Cake

Currently the fluent API is completely removed, but could be brought back into the settings classes if you still want it. Generally it should be much easier to extend to further npm commands this way, than as it was before, at the cost of a different API.

I'm happy to tweak it further if you have any feedback.  

Fixes #21 and #22.

### CODE REVIEW

**Comment by pascalberger** [comment]
@Philo Do you have interested in going into this direction? I would like to add some additional features to this package but they are depending on if you want to proceed with this PR or not.

**Comment by Philo** [comment]
@pascalberger - having just returned from holiday and having a bit of a crazy work/home schedule recently I've been horribly neglecting this repo (and others).

As briefly discussed previously, I think reworking the plugin to better follow the conventions used across other cake plugins is the right way to go.  While on one hand I would like to avoid simply introducing breaking changes between 2 versions I think that is probably the cleanest way, will just need to be very clear in docs and release notes about the changes.

I have also been thinking about whether there is an opportunity here to build a re-useable core library for node package execution, this could then to shared across various cake addins:

* Cake.Npm
* Cake.Gulp
* Cake.Webpack

I'm going to try and devote some time at the end of the week and hopefully the weekend to at the very least make a plan, discuss and integrate all the good suggestions and work from others (like yourself).

**Comment by pascalberger** [comment]
@Philo Absolutely no problem! Everyone has its private life

I also thought about creating a reusable library for running npm packages. Since eg. Cake.Gulp and Cake.Yarn have copied most of their code.

Just let me know if I should change anything in this PR (or you should be able to do changes yourself on my branch). Just if you plan to merge these changes, I would advice to do it before merging any other PRs, since it rewrite most of the code and it would be easier to adopt the other PRs to these changes than vice versa. 

As soon as this issue is cleared, I plan to add some additional features to Cake.Npm since I would use it to pack and publish some internal packages, which it currently doesn't support.

**Comment by Philo** [comment]
Definitely, depending on how my week goes and since you appear to be quite a heavy user of Cake.Npm would you be interested in helping out with contributor access to the repo?

My original reason for creating the package was to integrate the front end work that is done by other teams within my company.  So while I have a passing/working knowledge of Npm I cant say I am a power user.  It would be good to have another person aboard to help steer the direction of the plugin in areas I'm less familiar with?

**Comment by pascalberger** [comment]
@Philo Sure, would be happy to help

**Comment by Philo** [comment]
Fantastic, thanks

**Comment by gep13** [comment]
@Philo @pascalberger let me know how you want to proceed with this.  I would need to be the one to provide @pascalberger access to this repo, so that it all flows through to AppVeyor, etc.

You both happy for me to add @pascalberger into this repo?

**Comment by Philo** [comment]
Yes @gep13

**Comment by pascalberger** [comment]
@gep13 Yes, OK for me

**Comment by gep13** [comment]
@pascalberger I have given you access to the Cake.Npm repository.  Let me know if you need/want access to any of the others too.

**Comment by pascalberger** [comment]
@gep13 Would it be possible to also get access to the AppVeyor settings for Cake.Npm?

**Comment by gep13** [comment]
@pascalberger dang it! Try now.

**Comment by pascalberger** [comment]
@gep13 Thanks! Works now

File: src/Cake.Npm/NpmAliases.cs [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Provides a wrapper around Npm functionality within a Cake build script
+    /// </summary>
+    [CakeAliasCategory("Npm")]
+    [CakeNamespaceImport("Cake.Npm.Install")]
+    [CakeNamespaceImport("Cake.Npm.Pack")]
+    [CakeNamespaceImport("Cake.Npm.RunScript")]
+    public static class NpmAliases
```
Comment (Philo) [inline src/Cake.Npm/NpmAliases.cs]: @pascalberger - a question, and the answer may well be related to how cake handles aliases, but....

Would it be better to split the individual Npm commands (Install, Pack, RunScript, etc) into separate alias classes such as `NpmInstallAlias`, `NpmPackAlias` etc?

Comment (pascalberger) [inline src/Cake.Npm/NpmAliases.cs]: @Philo Splitting would make sense IMHO

Comment (Philo) [inline src/Cake.Npm/NpmAliases.cs]: I'm looking at the merge now, so I can do that.  Also looking to update the basic readme and `usage.cake` information.

**Comment by Philo** [comment]
I think I will initially push this as a prerelease version of `0.9.0` since it has specific breaking changes and allow consumers to test and migrate without risking breaking any CI builds that automatically pull the latest stable build of the addin.

**Comment by Philo** [comment]
The beta version of 0.9.0 is available on nuget now:

https://github.com/cake-contrib/Cake.Npm/releases/tag/0.9.0-beta.11

This incorporates all your updates from this PR.

**Pull request merged**